bind.go

```diff
@@ -103,7 +103,23 @@ func (b *DefaultBinder) bindData(ptr interface{}, data map[string][]string, tag
 				continue
 			}
 		}
+
 		inputValue, exists := data[inputFieldName]
+		if !exists {
+			// Go json.Unmarshal supports case insensitive binding.  However the
+			// url params are bound case sensitive which is inconsistent.  To
+			// fix this we must check all of the map values in a
+			// case-insensitive search.
+			inputFieldName = strings.ToLower(inputFieldName)
+			for k, v := range data {
```

Comment:
Instead of iterating the entire map, can you check for lowercase key existence? I know this won't work if your key is `Name`.

/cc @Carpetsmoker

Reply:
I'm not sure that gives us any real benefit to be honest.  I can write a benchmark if you really care, but the point is that it needs to be case insensitive, not just matching upper and lower case.  Additionally at worst this will be a loop over a few keys/values which is not going to perform all that different from a map lookup.

Reply:
@ready4god2513 Got it, merged.